AsyncNats/NatsServerPool.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace EightyDecibel.AsyncNats
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    internal class NatsServerPool
+    {
+
+        public List<NatsHost> Servers => new List<NatsHost>(_servers.Concat(_discoveredServers));
+
+        List<NatsHost> _servers = new List<NatsHost>();
+        List<NatsHost> _discoveredServers = new List<NatsHost>();
+        Random _random = new Random();
+        NatsHost? _lastSelectedServer = null;
+        INatsOptions _options;
+
+        object _sync = new object();
+
+        public NatsServerPool(INatsOptions options)
+        {
+
+#pragma warning disable CS0618 // Type or member is obsolete

```

Comment:
I'm fine with removing the single Server option and marking it as a breaking change. It's not a huge change for people to make in their code.

Reply:
ok